plugins/module_utils/network/sonic/config/dhcp_snooping/dhcp_snooping.py

```diff
@@ -0,0 +1,595 @@
+            if afi_config.get('verify_mac') is False:
+                requests.extend(self.get_delete_verify_mac_request(afi_config))
+            if afi_config.get('vlans') is not None and afi_config.get('vlans') != []:
+                requests.extend(self.get_delete_vlans_requests(afi_config))
+            if afi_config.get('trusted') is not None and afi_config.get('trusted') != []:
+                requests.extend(self.get_delete_trusted_requests(afi_config))
+            if afi_config.get('source_bindings') is not None and afi_config.get('source_bindings') != []:
+                requests.extend(self.get_delete_all_source_bindings_request())
+        return requests
+
+    def get_delete_specific_afi_fields_requests(self, want_afi, have_afi):
+        '''creates and builds list of requests for deleting some fields of afi config, depending on if
+        it is specified in playbook and matches with current config'''
+        requests = []
+
+        if want_afi.get('enabled') is True and have_afi.get('enabled') is True:
+            # only need to send a request if want from playbook is set to non default value and the setting currently configured is non default
+            requests.extend(self.get_delete_enabled_request(want_afi))
+        if want_afi.get('verify_mac') is False and have_afi.get('verify_mac') is False:
+            requests.extend(self.get_delete_verify_mac_request(want_afi))
+        if want_afi.get('vlans') is not None and have_afi.get('vlans') is not None and have_afi.get("vlans") != []:
+            # gathering list of vlans to be deleted. this section also handles cases where empty list of vlans is passed in
+            # which means delete all vlans
+            to_delete_vlans = have_afi["vlans"]
+            if len(want_afi["vlans"]) > 0:
+                to_delete_vlans = want_afi["vlans"]
+            # removing vlans that don't exist on device
+            for vlan in list(to_delete_vlans):
+                if vlan not in have_afi["vlans"]:
+                    to_delete_vlans.remove(vlan)
+            to_delete = {"afi": want_afi["afi"], "vlans": to_delete_vlans}
+            if len(to_delete["vlans"]):
+                requests.extend(self.get_delete_vlans_requests(to_delete))
+        if want_afi.get('trusted') is not None and have_afi.get('trusted') is not None and have_afi.get('trusted') != []:
+            # gathering list of interfaces to be deleted. this section also handles cases where empty list of interfaces is passed in which
+            #  means delete all trusted interfaces
+            to_delete_trusted = have_afi["trusted"]
+            if len(want_afi["trusted"]) > 0:
+                to_delete_trusted = want_afi["trusted"]
+            # removing interfaces that don't exist on device
+            for intf in list(to_delete_trusted):
+                if intf not in have_afi["trusted"]:
+                    to_delete_trusted.remove(intf)
```

Comment:
Interfaces check is required only specific interfaces are provided in 'want'
```suggestion
            to_delete_trusted = have_afi["trusted"]
            if len(want_afi["trusted"]) > 0:
                to_delete_trusted = want_afi["trusted"]
                # removing interfaces that don't exist on device
                for intf in list(to_delete_trusted):
                    if intf not in have_afi["trusted"]:
                        to_delete_trusted.remove(intf)
```